### PR TITLE
Fix epoch event

### PR DESCRIPTION
**Pull request by orangetin**
Pass `--epoch-steps` to indicate how many steps bw each epoch

### CODE REVIEW

File: training/dist_clm_train.py [hunk 1]
```diff
@@ -292,6 +292,7 @@ def main():
     parser.add_argument('--job-id', 
                         type=str, default="0", metavar='S',
                         help='an uuid')
+    parser.add_argument('--epoch-steps', type=int, default=0, help='-')
```
Comment (justusc) [inline training/dist_clm_train.py:295]: We should compute epoch steps. This conflicts with the option that specifies the number of steps.

**Review by justusc — CHANGES_REQUESTED** [review]

File: training/dist_clm_train.py (hunk 1 above)
Comment (orangetin) [reply training/dist_clm_train.py:295]: we'd need num-examples which we don't have

Comment (justusc) [reply training/dist_clm_train.py:295]: You will need to get the number of examples from train_data_loader. You may need to add a function to get the number of examples.

Comment (justusc) [reply training/dist_clm_train.py:295]: This probably works.

Comment (justusc) [reply training/dist_clm_train.py:295]: https://github.com/togethercomputer/OpenChatKit/blob/main/training/dist_clm_train.py#L347

Comment (orangetin) [reply training/dist_clm_train.py]: yup. That's what I used. Fixed in new commit.

File: training/dist_clm_train.py [hunk 2]
```diff
@@ -345,6 +345,8 @@ def main():
         
     if args.total_steps is None:
         args.total_steps = len(train_data_loader)
+
+    args.epoch_steps = int(len(train_data_loader) / args.batch_size)
```
Comment (justusc) [inline training/dist_clm_train.py:349]: I don't think we should be setting properties on args that are not passed in via CLI. Can we compute this value in `train_loop`?

Comment (orangetin) [reply training/dist_clm_train.py:349]: k

Comment (orangetin) [reply training/dist_clm_train.py]: fixed

**Review by justusc — CHANGES_REQUESTED** [review]
The way you implemented the check we won't get events if `len(train_data_loader) < args.batch_size`.

**Review by justusc — APPROVED** [review]

**Review by Scutua — APPROVED** [review]

**Pull request merged**